app/models/structure.rb

```diff
@@ -27,10 +27,11 @@ class Structure < ApplicationRecord
   after_validation :geocode, if: ->(s) { s.code_postal.present? and s.code_postal_changed? }
 
   scope :joins_evaluations_et_groupe, lambda {
-    joins('INNER JOIN comptes ON structures.id = comptes.structure_id')
-      .joins('INNER JOIN campagnes ON comptes.id = campagnes.compte_id')
-      .joins('INNER JOIN evaluations ON campagnes.id = evaluations.campagne_id')
+    joins('LEFT OUTER JOIN comptes ON structures.id = comptes.structure_id')
+      .joins('LEFT OUTER JOIN campagnes ON comptes.id = campagnes.compte_id')
+      .joins('LEFT OUTER JOIN evaluations ON campagnes.id = evaluations.campagne_id')
```

Comment:
effectivement ce changement me parait une bonne idée. N'écrirait-on pas un test pour le valider ?

Reply:
tu veux tester ça comment ?

Reply:
ben, avec mon clavier ! (rire)

Il me semble que le `LEFT OUTER` permet de retourner les structures qui n'ont pas de compte, les comptes qui n'ont pas de campagnes et les campagnes qui n'ont pas d'évaluations. Il est peut-être possible de vérifier que la liste des structures contient bien les structures avec zero évaluations ?

Reply:
ok alors écrivons des tests d'intégration

Reply:
Qu'est ce qu'on fait de cette PR ?
Cela fait 14 jours qu'elle a été posée et elle reste bloqué.

Personnellement, j'ai le sentiment qu'écrire ces tests d'intégration provoquera de nouvelle discussion car le scope de test me parait très large (et un peu flou pour moi), ce qui provoquera inévitablement une suite de test plus longue. 
Ce dernier point engendrera de nouvelle discussion .. (serpent qui se mort la queue).

Pour débloquer la situation, j'aimerais que l'on prenne une discussion sur ce que l'on fait : 
1. Rien
2. On définit exactement les tests que l'on veut

Reply:
On prend le cas d'une structure qui n'a pas d'évaluation
pareil pour chaque scope